crates/bevy_reflect/src/from_reflect.rs

```diff
@@ -88,25 +90,28 @@ pub trait FromReflect: Reflect + Sized {
 /// let registration = registry.get_with_name(reflected.type_name()).unwrap();
 /// let rfr = registration.data::<ReflectFromReflect>().unwrap();
 ///
-/// let concrete: Box<dyn Reflect> = rfr.from_reflect(&reflected).unwrap();
+/// let concrete: Box<dyn PartialReflect> = rfr.from_reflect(&reflected).unwrap();
 ///
-/// assert_eq!(Foo(123), concrete.take::<Foo>().unwrap());
+/// assert_eq!(Foo(123), concrete.try_take::<Foo>().unwrap());
 /// ```
 ///
 /// [`DynamicStruct`]: crate::DynamicStruct
 /// [`DynamicEnum`]: crate::DynamicEnum
 #[derive(Clone)]
 pub struct ReflectFromReflect {
-    from_reflect: fn(&dyn Reflect) -> Option<Box<dyn Reflect>>,
+    from_reflect: fn(&dyn PartialReflect) -> Option<Box<dyn PartialReflect>>,
```

Comment:
This should return `Option<Box<dyn Reflect>>` since it is the erased version of `Option<Self>`, and `Self` must implement `Reflect` since it is required by `FromReflect`.